src/scripts/Player.ts

```diff
@@ -22,28 +22,14 @@ class Player {
         this._region = ko.observable(savedPlayer._region);
         if (MapHelper.validRoute(savedPlayer._route, savedPlayer._region)) {
             this._route = ko.observable(savedPlayer._route);
+            this._town = ko.observable(TownList['Pallet Town']);
+            this._currentTown = ko.observable('Pallet Town');
         } else {
-            switch (savedPlayer._region) {
-                case 0:
-                    this._route = ko.observable(1);
-                    break;
-                case 1:
-                    this._route = ko.observable(29);
-                    break;
-                case 2:
-                    this._route = ko.observable(101);
-                    break;
-                case 3:
-                    this._route = ko.observable(201);
-                    break;
-                default:
-                    this._route = ko.observable(1);
-                    this._region = ko.observable(GameConstants.Region.kanto);
-            }
+            this._route = ko.observable(0);
+            const town = savedPlayer._currentTown || GameConstants.StartingTowns[savedPlayer._region];
```

Comment:
new games fall into this `else` because they have no `_route` or `_region`, but this change has removed the default of `1, kanto`, causing new games to crash on loading the page.